### PR TITLE
Add neighborhoods to typeahead list and prioritize in dropdown

### PR DESCRIPTION
**Pull request by CloudNiner**
typeahead-neighborhood
Use custom template for typeahead item

Display "Neighborhood" text on right if item is a neighborhood
result.

### CODE REVIEW

**Comment by CloudNiner** [comment]
@jfrankl @jbranigan should take a look as well

File: app/scripts/views/search/location-search-directive.js [hunk 1]
```diff
@@ -12,39 +12,83 @@
...
+            $log.debug(result);
+            // Address result
+            if (result.magicKey !== undefined) {
+                Geocoder.search(result.text, result.magicKey, options)
+                .then(function (results) {
+                    if (results && results.length) {
+                        var result = results[0];
+                        var latLng = L.latLng(result.feature.geometry.y, result.feature.geometry.x);
+                        if (searchBounds.contains(latLng)) {
+                            // TODO: Determine if neighborhood or address result and properly
```
Comment (flibbertigibbet) [inline app/scripts/views/search/location-search-directive.js]: Aren't neighborhoods handled in the `else if (results.centroid)` block below?

**Comment by flibbertigibbet** [comment]
LGTM.  If the results are all neighborhood names or short, will the right-aligned text still be alright?

File: app/scripts/views/search/location-search-directive.js (hunk 1 above)
Comment (CloudNiner) [inline app/scripts/views/search/location-search-directive.js]: You're right, this todo can come out

**Comment by jfrankl** [comment]
+1 looks good

**Pull request merged**